### PR TITLE
Fill properties from file

### PR DESCRIPTION
**Pull request by nextinteractivemobile**
This PR let us provide a file path to fill the properties (description, changelog, groups). 

If a file path is provided and the file content exists, the content is used to fill the property otherwise the value is taken.

### CODE REVIEW

File: src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java [hunk 1]
```diff
@@ -143,9 +143,11 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
...
         }
         
-    	String descriptionToSend = evaluateField(description, build, listener);
-    	String changelogToSend = evaluateField(changelog, build, listener);
-    	String groupsToSend = evaluateField(groups, build, listener);
+    	String descriptionToSend = evaluateField(description, build, listener, rootDir);
+
+    	String changelogToSend = evaluateField(changelog, build, listener, rootDir);
+
+        String groupsToSend = evaluateField(groups, build, listener, rootDir);
```
Comment (chrsvl) [inline src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java]: Could you remove the line breaks and re-indent properly line 150 please?

File: src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java [hunk 2]
```diff
@@ -177,9 +179,18 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
...
     }
 
-	private String evaluateField(String field, AbstractBuild build, BuildListener listener) throws IOException, InterruptedException {
+	private String evaluateField(String field, AbstractBuild build, BuildListener listener, FilePath rootDirectory) throws IOException, InterruptedException {
     	EnvVars vars = build.getEnvironment(listener);
-    	return vars.expand(field);
+    	String fieldValue = vars.expand(field);
+
+        if (!fieldValue.isEmpty()) {
+            //If the filed if a file path and it exists, we grab the content, otherwise the field is taken as is.
```
Comment (chrsvl) [inline src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java]: Small typo here: `If the filed is a file`

**Comment by antoninbiret** [comment]
Hi @chrsvl. 

I made some changes on the PR. Hope it's ok now.

File: src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java [hunk 3]
```diff
@@ -177,9 +177,18 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
...
     }
 
-	private String evaluateField(String field, AbstractBuild build, BuildListener listener) throws IOException, InterruptedException {
+	private String evaluateField(String field, AbstractBuild build, BuildListener listener, FilePath rootDirectory) throws IOException, InterruptedException {
     	EnvVars vars = build.getEnvironment(listener);
-    	return vars.expand(field);
+    	String fieldValue = vars.expand(field);
+
+        if (!fieldValue.isEmpty()) {
+            //If the field if a file path and it exists, we grab the content, otherwise the field is taken as is.
```
Comment (chrsvl) [inline src/main/java/org/jenkins/plugins/appaloosa/AppaloosaPublisher.java]: Could you please fix the typo here :) ?
"If the field ***is*** a file"

**Comment by chrsvl** [comment]
Thank you very much @antoninbiret !

**Pull request merged**